platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/pool/DefaultTransactionPool.java

```diff
@@ -42,7 +43,7 @@ public DefaultTransactionPool(@NonNull final TransactionPoolNexus transactionPoo
      * {@inheritDoc}
      */
     @Override
-    public void submitSystemTransaction(@NonNull final ConsensusTransactionImpl transaction) {
+    public void submitSystemTransaction(@NonNull final OneOf<PayloadOneOfType> transaction) {
```

Comment:
maybe we should check if its actually a system transaction. Or maybe its even better if we just allow the class `StateSignaturePayload` to be submitted